sources/packages/backend/libs/services/src/disbursement-overaward/disbursement-overaward.service.ts

```diff
@@ -136,16 +140,36 @@ export class DisbursementOverawardService {
   async addManualOveraward(
     awardValueCode: string,
     overawardValue: number,
+    overawardNotes: string,
     studentId: number,
     auditUserId: number,
   ): Promise<DisbursementOveraward> {
-    const overawardManualRecord = new DisbursementOveraward();
-    overawardManualRecord.creator = { id: auditUserId } as User;
-    overawardManualRecord.disbursementValueCode = awardValueCode;
-    overawardManualRecord.overawardValue = overawardValue;
-    overawardManualRecord.originType =
-      DisbursementOverawardOriginType.ManualRecord;
-    overawardManualRecord.student = { id: studentId } as Student;
-    return this.disbursementOverawardRepo.save(overawardManualRecord);
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        // Create the note for the overaward and associate the note with the student.
+        const noteEntity = await this.noteSharedService.createStudentNote(
+          studentId,
+          NoteType.Overaward,
+          overawardNotes,
+          auditUserId,
+          transactionalEntityManager,
+        );
+
+        // Save disbursement overaward record.
+        const overawardManualRecord = new DisbursementOveraward();
+        overawardManualRecord.creator = { id: auditUserId } as User;
```

Comment:
Audit user could be defined as const `const auditUser =  { id: auditUserId } as User;` and used at line 160 and 167.
And likewise current time can be created as const `const now = new Date()` and be assigned to `overawardManualRecord.addedDate` and `overawardManualRecord.createdAt` so that the both the time perfectly match. (We do this on other places as well)

Reply:
Regarding the `auditUser` is nice to have a const for it. 
About the dates I think it is even better to have different dates because we can assure that the date that is `addedDate` is the date that it was supposed to be there, and not got from `createdAt`.

Reply:
At this given context createdAt and AddedBy are populated by same action. Technically they are suppose to be same timestamp,but it is just that createdAt is populated by DB milliseconds later than addedBy is assigned they have a different value in them.

Btw, it is not a blocker but good to have.